app.rb

```diff
@@ -66,14 +55,6 @@ def create_person
     end
   end
 
-  def create_teacher(name, age)
-    print 'Specialization: '
-    specialization = gets.chomp
-    person = Teacher.new(age, specialization, name)
-    @people << person
-    puts 'Teacher created successfully.'
-  end
-
   def create_student(name, age)
     print 'Has parent permission? [Y/N] : '
     permission = gets.chomp
```

Comment:
- [ ] Kindly consider creating a separate module for requesting input or validating input and then you can call the method wherever you need the user input in any method of app.rb. this way you will be following SOLID principles.